stat/distuv/laplace_test.go

```diff
@@ -56,3 +58,40 @@ func TestLaplaceProb(t *testing.T) {
 	}
 	testDistributionProbs(t, Laplace{Mu: 0, Scale: 1}, "Laplace", pts)
 }
+func TestLaplaceFit(t *testing.T) {
+	cases := []struct {
+		samples   []float64
+		weights   []float64
+		wantMu    float64
+		wantScale float64
+	}{
+		{
+			samples:   []float64{10, 1, 1},
+			weights:   nil,
+			wantMu:    1,
+			wantScale: 3,
+		},
+		{
+			samples:   []float64{10, 1, 1},
+			weights:   []float64{10, 10, 10},
+			wantMu:    1,
+			wantScale: 3,
+		},
+		{
+			samples:   []float64{10, 1, 1},
+			weights:   []float64{0, 1, 1},
+			wantMu:    1,
+			wantScale: 0,
+		},
+	}
+	for i, test := range cases {
+		d := Laplace{}
+		d.Fit(test.samples, test.weights)
+		if !floats.EqualWithinAbsOrRel(d.Mu, test.wantMu, 1e-10, 1e-10) {
+			t.Errorf("unexpected location result for test %d: got:%f, want:%f", i, d.Mu, test.wantMu)
+		}
+		if !floats.EqualWithinAbsOrRel(d.Scale, test.wantScale, 1e-10, 1e-10) {
+			t.Errorf("unexpected scale result for test %d: got:%f, want:%f", i, d.Scale, test.wantScale)
+		}
+	}
```

Comment:
Could you also add a randomized test? Something like

```
nSamples := 10000
src := rand.New(rand.NewSource(1))
l := Laplace {
    Mu: 3,
    Scale: 5,
    Src: src,
}
samples := make([]float64, nSamples)
for i := range samples {
    samples[i] = l.Rand()
}
l2 := Laplace{}
l2.Fit(samples, nil)
// Compare mu and scale to within 1e-2 or 1e-3
```

Reply:
done.